src/hal/user_comps/mqtt-publisher.py

```diff
@@ -84,9 +84,18 @@ def create_pins(self, keys):
     def update_mqtt(self):
         "Run the endless loop fetching pin data and sending it to MQTT."
         running = True
+        missing = {}
         while hal.component_exists('halui') and running:
             data = {}
             if self.hal['enable']:
+                for key in self.keys:
+                    try:
+                         data[key] = hal.get_value(key)
+                    except RuntimeError as e:
```

Comment:
Is RuntimeError really the correct exception from get_value(key)?  It seem overly broad.

Reply:
Fair comment, AttributeError seems to work just fine.

Reply:
was testing wrong code, AttributeError does not catch the error thrown.